tests/console/gdb.pm

```diff
@@ -0,0 +1,81 @@
+# SUSE's openQA tests
+#
+# Copyright (C) 2019 SUSE LLC
+#
+# Copying and distribution of this file, with or without modification,
+# are permitted in any medium without royalty provided the copyright
+# notice and this notice are preserved. This file is offered as-is,
+# without any warranty.
+# Summary: Basic GDB test. (Breakpoints/backtraces/attaching)
+# Maintainer: apappas@suse.de
+
+use base 'consoletest';
+use strict;
+use warnings;
+use testapi;
+use utils 'zypper_call';
+
+sub wait_serial_or_croak {
+    my $feedback = shift;
+
+    my $e = wait_serial($feedback, 10);
+    if (!defined $e) {
+        croak("Unexpected serial output");
+    }
+}
+
+
+sub run {
+    #Setup console for text feedback.
+    my ($self) = @_;
+    $self->select_serial_terminal();
+
+    zypper_call('in gcc glibc-devel gdb');    #Install test depedencies.
+
+    #Test Case 1
+    assert_script_run("curl -O " . data_url('gdb/test1.c'));
+    assert_script_run("gcc -g -std=c99 test1.c -o test1");
+    script_run("gdb test1");
```

Comment:
This can never have worked - and your verification run showed this fail as well: http://10.161.229.249/tests/81#step/gdb/12

`script_run` expects the command to return but *gdb* doesn't return but show a prompt so you will need to use `type_string("foo\n")` to start *gdb*.

It is now failing on osd: https://openqa.suse.de/tests/2959947#step/gdb/8

Reply:
This is my first test so could you elaborate? What do you mean, this could have never worked?
It starts gdb with test1 as you can see in http://10.161.229.249/tests/81/file/serial_terminal.txt (search for `Starting program: /root/test1`). 

Do you mean that the serial console expects it to return, and gives an error on that step, but not the whole test? Because it wouldn't fail the whole test, and I use wait_serial to test for expected output, I ignored this particular failure. Should I avoid that? 

Of course I'll change it to send_key though, but I still want to learn.